test/input/yaml/CustomInput_loop_da.yaml

```diff
@@ -0,0 +1,159 @@
+    compute_kernel: V02-structured  # OPTIONS: "V02-caching", "V02-structured-obj", "V02-structured", "V02-diffusive-dummy"
+    assume_short_ts: true  # use the previous timestep value for both current and previous flow.
+    return_courant: false  # WARNING: true will only work with compute_kernel "V02-caching", therefore not currently compatible with simulation for waterbodies.
+    #WRF-Hydro restart files
+    restart_parameters:
+        #WRF-Hydro channels restart file
+        wrf_hydro_channel_restart_file: "/glade/scratch/jamesmcc/tmp_florence_run_nudging/run_nudging_nwm_channel-only/HYDRO_RST.2018-09-18_00:00_DOMAIN1"
+        #WRF-Hydro channels ID crosswalk file
+        wrf_hydro_channel_ID_crosswalk_file: "/glade/scratch/jamesmcc/tmp_florence_run_nudging/forecasts/cast_2018091800/NWM/DOMAIN/Route_Link.nc"
+        wrf_hydro_channel_ID_crosswalk_file_field_name: link
+        wrf_hydro_channel_restart_upstream_flow_field_name: qlink1
+        wrf_hydro_channel_restart_downstream_flow_field_name: qlink2
+        wrf_hydro_channel_restart_depth_flow_field_name: hlink
+        #WRF-Hydro waterbodies restart file
+        wrf_hydro_waterbody_restart_file: "/glade/scratch/jamesmcc/tmp_florence_run_nudging/run_nudging_nwm_channel-only/HYDRO_RST.2018-09-18_00:00_DOMAIN1"
+        #WRF-Hydro waterbody ID crosswalk file
+        wrf_hydro_waterbody_ID_crosswalk_file: "/glade/scratch/jamesmcc/tmp_florence_run_nudging/forecasts/cast_2018091800/NWM/DOMAIN/LAKEPARM.nc"
+        wrf_hydro_waterbody_ID_crosswalk_file_field_name: lake_id
+        #WRF-Hydro waterbody crosswalk filter file
+        wrf_hydro_waterbody_crosswalk_filter_file: "/glade/scratch/jamesmcc/tmp_florence_run_nudging/forecasts/cast_2018091800/NWM/DOMAIN/Route_Link.nc"
+        wrf_hydro_waterbody_crosswalk_filter_file_field_name: NHDWaterbodyComID
+    #Qlateral forcing values
+    forcing_parameters:
+        # coastal_boundary_elev_data: "../../test/input/geo/coastal_inputs/staout_1"
+        # coastal_ncdf: "../../test/input/geo/coastal_inputs/SandySample.nc"
+        # split_forcing: true
+        qts_subdivisions: 12  # number of timesteps per forcing (qlateral) timestep.
+        dt: 300  # default timestep length, seconds
+        qlat_input_folder: "/glade/scratch/jamesmcc/tmp_florence_run_nudging/forecasts/cast_2018091800"
+        qlat_file_index_col: feature_id
+        qlat_file_value_col: q_lateral
+        # TODO: Presently, an explicit definition of these sets
+        # means the least ambiguity in how we intend to set things up.
+        # a more sophisticated algorithm could accept a glob list
+        # and break that up according to some `qlat_forcing_configuration`
+        # but for now, we'll use this.
+        qlat_forcing_sets:
+            qlat_file_range:
+                qlat_start_file: "201809180100.CHRTOUT_DOMAIN1"
+                qlat_end_file: "201809180600.CHRTOUT_DOMAIN1"
+        qlat_run_block_size: 4
+            # - qlat_files:
```

Comment:
@BrianAvant-NOAA you can uncomment the blocks to run code that way, but over longer periods of time, it will make sense to do it dynamically with the file range in line 94.